--- a/src/effects/overview/qml/DesktopBar.qml
+++ b/src/effects/overview/qml/DesktopBar.qml
@@ -67,7 +67,11 @@ Item {
                     Keys.onRightPressed: nextItemInFocusChain(!LayoutMirroring.enabled).forceActiveFocus(Qt.TabFocusReason);
 
                     function activate() {
-                        thumbnail.state = "scaled";
+                        if (KWinComponents.Workspace.currentVirtualDesktop !== delegate.desktop) {
+                            KWinComponents.Workspace.currentVirtualDesktop = delegate.desktop;
+                        } else {
+                            effect.deactivate();
+                        }
                     }
 
                     function remove() {
@@ -98,48 +102,16 @@ Item {
                         DesktopView {
                             id: thumbnail
 
-                            property bool scaled: state === "scaled"
-
                             width: targetScreen.geometry.width
                             height: targetScreen.geometry.height
-                            visible: scaled
+                            visible: true
                             clientModel: bar.clientModel
                             desktop: delegate.desktop
                             scale: bar.desktopHeight / targetScreen.geometry.height
                             transformOrigin: Item.TopLeft
 
-                            // Disable the item layer while being scaled.
-                            layer.enabled: !scaled
+                            layer.enabled: true
                             layer.textureSize: Qt.size(bar.desktopWidth, bar.desktopHeight)
-
-                            states: State {
-                                name: "scaled"
-                                ParentChange {
-                                    target: thumbnail
-                                    parent: container
-                                    x: 0
-                                    y: 0
-                                    scale: 1
-                                }
-                            }
-
-                            transitions: Transition {
-                                SequentialAnimation {
-                                    ParentAnimation {
-                                        NumberAnimation {
-                                            properties: "x,y,scale"
-                                            duration: effect.animationDuration
-                                            easing.type: Easing.OutCubic
-                                        }
-                                    }
-                                    ScriptAction {
-                                        script: {
-                                            KWinComponents.Workspace.currentVirtualDesktop = delegate.desktop;
-                                            effect.quickDeactivate();
-                                        }
-                                    }
-                                }
-                            }
                         }
 
                         OpacityMask {
@@ -156,7 +128,7 @@ Item {
                             border.width: 2
                             border.color: PlasmaCore.ColorScope.highlightColor
                             opacity: dropArea.containsDrag ? 0.5 : 1.0
-                            visible: !thumbnail.scaled && (delegate.activeFocus || dropArea.containsDrag || bar.selectedDesktop === delegate.desktop)
+                            visible: (delegate.activeFocus || dropArea.containsDrag || bar.selectedDesktop === delegate.desktop)
                         }
 
                         MouseArea {
