--- a/src/effects/overview/qml/DesktopBar.qml
+++ b/src/effects/overview/qml/DesktopBar.qml
@@ -179,7 +179,6 @@
                         }
 
                         Loader {
-                            LayoutMirroring.enabled: Qt.application.layoutDirection === Qt.RightToLeft
                             active: !heap.dragActive && (hoverHandler.hovered || Kirigami.Settings.tabletMode || Kirigami.Settings.hasTransientTouchInput) && desktopRepeater.count > 1
                             anchors.right: parent.right
                             anchors.top: parent.top
--- a/src/effects/overview/qml/ScreenView.qml
+++ b/src/effects/overview/qml/ScreenView.qml
@@ -18,6 +18,8 @@
 FocusScope {
     id: container
     focus: true
+    LayoutMirroring.enabled: Qt.application.layoutDirection === Qt.RightToLeft
+    LayoutMirroring.childrenInherit: true
 
     required property QtObject effect
     required property QtObject targetScreen
